Extension/src/LanguageServer/client.ts

```diff
@@ -1043,6 +1052,61 @@ export class DefaultClient implements Client {
                         }
                     }
 
+                    // Semantic token types are identified by indexes in this list of types, in the legend.
+                    let tokenTypesLegend: string[] = [];
+                    for (let e in SemanticTokenTypes) {
+                        // An enum is actually a set of mappings from key <=> value.  Enumerate over only the names.
+                        // This allow us to represent the constants using an enum, which we can match in native code.
+                        if (isNaN(Number(e))) {
+                            tokenTypesLegend.push(e);
+                        }
+                    }
+                    // Semantic token modifiers are bit indexes corresponding to the indexes in this list of modifiers in the legend.
+                    let tokenModifiersLegend: string[] = [];
+                    for (let e in SemanticTokenModifiers) {
+                        if (isNaN(Number(e))) {
+                            tokenModifiersLegend.push(e);
+                        }
+                    }
+                    const legend: vscode.SemanticTokensLegend = new vscode.SemanticTokensLegend(tokenTypesLegend, tokenModifiersLegend);
+
+                    class SemanticTokensProvider implements vscode.DocumentSemanticTokensProvider {
```

Comment:
Is there a way to have `class SemanticTokensProvider` in a different file like the old colorization class? Or will this require some major refactoring in Client.ts?